File: conversions/semver/elide-semver.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-semver
#
# This demonstrates calling the TypeScript semver implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One semver implementation shared across TypeScript and Ruby
# - Consistent parse and compare semantic version numbers across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript semver ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# semver_module = Elide.require('./elide-semver.ts')

# Example: Basic usage
# result = semver_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses parse and compare semantic version numbers"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for version management, dependency resolution, release automation"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide semver (TypeScript)         │"
puts "│   conversions/semver/elide-semver.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same parse and compare semantic version numbers everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/semver.rb"
puts "  SEMVER = Elide.require('./elide-semver.ts')"

File: conversions/semver/elide-semver.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-semver

This demonstrates calling the TypeScript semver implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One semver implementation shared across TypeScript and Python
- Consistent parse and compare semantic version numbers across services
- No Python custom implementation needed
- Perfect for version management, dependency resolution, release automation
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# semver_module = require('./elide-semver.ts')

print("=== Python Calling TypeScript semver ===\n")

# Example: Basic usage
# result = semver_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses parse and compare semantic version numbers")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for version management, dependency resolution, release automation")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide semver (TypeScript)         │")
print("│   conversions/semver/elide-semver.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same parse and compare semantic version numbers everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  semver = require('./elide-semver.ts')")
print("  # Use it directly!")
